### PR TITLE
issue 380.2.1: remove extraneous pointers from StorageEngine API

### PR DESCRIPTION
**Pull request by vasiliy-arangodb**

### CODE REVIEW

File: arangod/RestHandler/RestUsersHandler.cpp [hunk 1]
```diff
@@ -181,33 +181,46 @@ void RestUsersHandler::generateDatabaseResult(auth::UserManager* um,
...
+          velocypack::ObjectBuilder b2(&data, "collections", true);
+
+          methods::Collections::enumerate(
+            &vocbase,
+            [&](LogicalCollection* c)->void {
+              lvl =
+                user.configuredCollectionAuthLevel(vocbase.name(), c->name());
+                data.add(
+                  c->name(),
+                  velocypack::Value(convertFromAuthLevel(lvl))
```
Comment (graetzer) [inline arangod/RestHandler/RestUsersHandler.cpp]: why change the` VPack*` to `velocypack::*` here ?

File: arangod/RestHandler/RestUsersHandler.cpp [hunk 2]
```diff
@@ -181,33 +181,46 @@ void RestUsersHandler::generateDatabaseResult(auth::UserManager* um,
...
-      } else {  // hide db's without access
-        auth::Level lvl = user.databaseAuthLevel(vocbase->name());
-        if (lvl >= auth::Level::RO) {
-          data.add(vocbase->name(), VPackValue(convertFromAuthLevel(lvl)));
+    DatabaseFeature::DATABASE->enumerateDatabases(
+      [&](TRI_vocbase_t& vocbase)->void {
+        if (full) {
+          auto lvl = user.configuredDBAuthLevel(vocbase.name());
+          std::string str = convertFromAuthLevel(lvl);
+          velocypack::ObjectBuilder b(&data, vocbase.name(), true);
```
Comment (graetzer) [inline arangod/RestHandler/RestUsersHandler.cpp]: why change `VPack*` to `velocypack::*` here ?

**Comment by vasiliy-arangodb** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr-linux/776/

File: arangod/RestHandler/RestUsersHandler.cpp (hunk 1 above)
Comment (vasiliy-arangodb) [reply arangod/RestHandler/RestUsersHandler.cpp]: it's not much longer and makes it clear which namespace the object refers to, also easier in IDEs to switch to the implementation

File: arangod/RestHandler/RestUsersHandler.cpp (hunk 2 above)
Comment (vasiliy-arangodb) [reply arangod/RestHandler/RestUsersHandler.cpp]: it's not much longer and makes it clear which namespace the object refers to, also easier in IDEs to switch to the implementation

**Comment by vasiliy-arangodb** [comment]
replaced with https://github.com/arangodb/arangodb/pull/5589

**Pull request closed without merging**